OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs

```diff
@@ -18,172 +18,303 @@ namespace OutOfSchool.WebApi.Services
+        public ChildService(IEntityRepository<Child> childRepository, IParentRepository parentRepository, ILogger<ChildService> logger, IStringLocalizer<SharedResource> localizer)
         {
-            this.localizer = localizer;
-            this.repository = repository;
-            this.logger = logger;
+            this.childRepository = childRepository ?? throw new ArgumentNullException(nameof(childRepository));
+            this.parentRepository = parentRepository ?? throw new ArgumentNullException(nameof(parentRepository));
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.localizer = localizer ?? throw new ArgumentNullException(nameof(localizer));
         }
 
         /// <inheritdoc/>
-        public async Task<ChildDto> Create(ChildDto dto)
+        public async Task<ChildDto> CreateChildForUser(ChildDto childDto, string userId)
         {
-            logger.LogInformation("Child creating was started.");
+            logger.LogDebug($"Started creation of a new child with {nameof(Child.ParentId)}:{childDto.ParentId}, {nameof(userId)}:{userId}.");
 
-            this.Check(dto);
+            this.ValidateChildDto(childDto);
 
-            var child = dto.ToDomain();
+            if (userId is null)
+            {
+                throw new ArgumentNullException($"{nameof(userId)}");
+            }
+
+            var parent = (await parentRepository.GetByFilter(p => p.UserId == userId).ConfigureAwait(false)).SingleOrDefault()
+                ?? throw new ArgumentException($"There is no Parent with {nameof(userId)}:{userId}", userId);
+
+            if (childDto.ParentId != parent.Id)
+            {
+                logger.LogWarning($"Prevented action! User:{userId} with {nameof(Child.ParentId)}:{parent.Id} was trying to create a new child with not his own {nameof(Child.ParentId)}:{childDto.ParentId}.");
+                childDto.ParentId = parent.Id;
+            }
 
-            var newChild = await repository.Create(child).ConfigureAwait(false);
+            childDto.Id = 0;
 
-            logger.LogInformation($"Child with Id = {newChild?.Id} created successfully.");
+            var newChild = await childRepository.Create(childDto.ToDomain()).ConfigureAwait(false);
+
+            logger.LogDebug($"Child with Id:{newChild?.Id} ({nameof(Child.ParentId)}:{newChild.ParentId}, {nameof(userId)}:{userId}) was created successfully.");
 
             return newChild.ToModel();
         }
 
         /// <inheritdoc/>
-        public async Task<IEnumerable<ChildDto>> GetAll()
+        public async Task<SearchResult<ChildDto>> GetAllWithOffsetFilterOrderedById(OffsetFilter offsetFilter)
         {
-            logger.LogInformation("Getting all Children started.");
+            logger.LogDebug($"Getting all Children started. Amount of children to take: {offsetFilter.Size}, skip first: {offsetFilter.From}.");
+
+            if (offsetFilter is null)
```

Comment:
if `offsetFilter is null` you'll get NRE in previous line with `logger.LogDebug`